backend/src/main/java/com/bakdata/conquery/resources/admin/rest/AdminDatasetProcessor.java

```diff
@@ -287,16 +285,30 @@ public void setStructure(Namespace namespace, StructureNode[] structure) {
 	public void addImport(Namespace namespace, InputStream inputStream) throws IOException {
 
 		ImportJob job = ImportJob.create(namespace, inputStream, config.getCluster().getEntityBucketSize(), sharedDictionaryLocks, config);
-
+		ImportsManager importsManager = ImportsManager.getInstance();
+		importsManager.addStartedImportJob(job.getImportId().toString(), job.getJobId());
 		namespace.getJobManager().addSlowJob(job);
 	}
 
+
+	public void updateImport(Import imp, Namespace namespace, InputStream inputStream) throws IOException {
+		ImportsManager importsManager = ImportsManager.getInstance();
+		if (importsManager.isJobRunning(imp.getId().toString())) {
```

Comment:
Ich denke es ist hier wäre IdMutex praktisch:

In `updateImport`, bzw `ImportJob.createImport` machst versuchst du den Mutex zu der ImportId zu aquirieren und am Ende des Jobs gibst du das Lock wieder frei.
Ich bin mir dann nur nicht ganz sicher wie die Reihenfolge der wartenden Threads ist.

Aber dann solltest du nicht einen ImportManager benötigen